src/notecard_platform.c

```diff
@@ -385,3 +495,45 @@ void notecard_platform_free(void *ptr)
 {
     free(ptr);
 }
+
+//=============================================================================
+// Mutex Hook Registration
+//=============================================================================
+
+void notecard_platform_register_mutex_hooks(void)
+{
+    // Register mutex hooks with note-c based on Kconfig settings
+#ifdef CONFIG_NOTECARD_I2C_MUTEX
+    // Both I2C and Notecard mutexes enabled
+    NoteSetFnMutex(notecard_platform_lock_i2c,
+                   notecard_platform_unlock_i2c,
+                   notecard_platform_lock_note,
+                   notecard_platform_unlock_note);
+#else
+    // Only Notecard mutex enabled (I2C mutex is optional)
+    NoteSetFnNoteMutex(notecard_platform_lock_note,
+                       notecard_platform_unlock_note);
+#endif
```

Comment:
I think you gain in clarity anything you lose in two calls.

```suggestion
    // Enable Notecard mutexes (I2C mutex is optional)
    NoteSetFnNoteMutex(notecard_platform_lock_note,
                       notecard_platform_unlock_note);

#ifdef CONFIG_NOTECARD_I2C_MUTEX
    // Enable I2C mutexes
    NoteSetFnI2CMutex(notecard_platform_lock_note,
                   notecard_platform_unlock_note);
#endif
```

Also, is the flag `CONFIG_NOTECARD_I2C_MUTEX` only set if you use the I2C interface, or will it also be set if you use UART?

Reply:
Good catch, it's the latter so I need to handle that

Reply:
The mutex is only created if notecard_platform_i2c_init() is called so it shouldnt be a problem if CONFIG_NOTECARD_I2C_MUTEX=y but the user is using UART